cdap-common/src/main/java/co/cask/cdap/common/resource/ResourceBalancerService.java

```diff
@@ -49,46 +47,44 @@
 import java.util.Collection;
 import java.util.Set;
 import java.util.concurrent.Executor;
-import javax.annotation.Nullable;
 
 /**
- * Metrics processor service that processes events from Kafka.
+ * A services that automatically balances resource assignments between its instances.
  */
-public final class KafkaMetricsProcessorService extends AbstractExecutionThreadService {
+public abstract class ResourceBalancerService extends AbstractExecutionThreadService {
```

Comment:
Since this class and sub-classes has nothing to "run", using `AbstractIdleService` instead (so that you don't need a completion and just block on it).